source/slang/slang-ir-inst-defs.h

```diff
@@ -291,6 +291,10 @@ INST(IndexedFieldKey, indexedFieldKey, 2, HOISTABLE)
 // A placeholder witness that ThisType implements the enclosing interface.
 // Used only in interface definitions.
 INST(ThisTypeWitness, thisTypeWitness, 1, 0)
+
+// A placeholder witness for the fact that two types are equal.
+INST(TypeEqualityWitness, TypeEqualityWitness, 2, HOISTABLE)
+
```

Comment:
When we get around to being more formal with our type system, we may find that we need more nuance here and a type equality witness isn't just a no-op "placeholder."

It really comes down to whether we are talking about *identical* types, or just *equivalent* types. If it is possible in our system to have two types `T` and `U` where each is a subtype of the other, but they are *not* identical, then those types are equivalent, and a witness to their equivalence would basically amount to a pair of witnesses: one for each of the subtype relations involved. Those subtype witnesses would in turn but usable for converting a value of type `T` into one of type `U`, and vice versa, which might not otherwise be possible for the IR codegen to synthesize.

Reply:
Because inheritance isn't really a thing right now in the language, such case won't arise. When we have general support of inheritance, then equivalence vs equality becomes relevant and we will need more formality about it.